dsp/window/cmd/leakage/leakage.go

```diff
@@ -0,0 +1,387 @@
+		ok:   func(_ float64) bool { return true },
+	},
+	"hann": {
+		name: func(_ float64) string { return "Hann" },
+		fn:   func(_ float64) func([]float64) []float64 { return window.Hann },
+		ok:   func(_ float64) bool { return true },
+	},
+	"bartletthann": {
+		name: func(_ float64) string { return "BartlettHann" },
+		fn:   func(_ float64) func([]float64) []float64 { return window.BartlettHann },
+		ok:   func(_ float64) bool { return true },
+	},
+	"hamming": {
+		name: func(_ float64) string { return "Hamming" },
+		fn:   func(_ float64) func([]float64) []float64 { return window.Hamming },
+		ok:   func(_ float64) bool { return true },
+	},
+	"blackman": {
+		name: func(_ float64) string { return "Blackman" },
+		fn:   func(_ float64) func([]float64) []float64 { return window.Blackman },
+		ok:   func(_ float64) bool { return true },
+	},
+	"blackmanharris": {
+		name: func(_ float64) string { return "BlackmanHarris" },
+		fn:   func(_ float64) func([]float64) []float64 { return window.BlackmanHarris },
+		ok:   func(_ float64) bool { return true },
+	},
+	"nuttall": {
+		name: func(_ float64) string { return "Nuttall" },
+		fn:   func(_ float64) func([]float64) []float64 { return window.Nuttall },
+		ok:   func(_ float64) bool { return true },
+	},
+	"blackmannuttall": {
+		name: func(_ float64) string { return "BlackmanNuttall" },
+		fn:   func(_ float64) func([]float64) []float64 { return window.BlackmanNuttall },
+		ok:   func(_ float64) bool { return true },
+	},
+	"flattop": {
+		name: func(_ float64) string { return "FlatTop" },
+		fn:   func(_ float64) func([]float64) []float64 { return window.FlatTop },
+		ok:   func(_ float64) bool { return true },
+	},
+	"gaussian": {
+		name: func(p float64) string { return fmt.Sprintf("Gaussian σ=%v", p) },
+		fn:   func(p float64) func([]float64) []float64 { return window.Gaussian{Sigma: p}.Transform },
+		ok:   func(p float64) bool { return !math.IsNaN(p) },
+	},
+	"tukey": {
+		name: func(p float64) string { return fmt.Sprintf("Tukey α=%v", p) },
+		fn:   func(p float64) func([]float64) []float64 { return window.Tukey{Alpha: p}.Transform },
+		ok:   func(p float64) bool { return !math.IsNaN(p) },
+	},
+}
+
+type builtin struct {
+	name func(float64) string
+	fn   func(float64) func([]float64) []float64
+	ok   func(float64) bool
+}
+
+func main() {
+	name := flag.String("window", "", "specify a built-in window name")
+	param := flag.Float64("param", math.NaN(), "specify parameter for parametric windows")
+	symm := flag.Bool("symm", true, "specify whether the window is symmetric")
+	n := flag.Int("n", 20, "specify window length")
+	m := flag.Int("m", 1000, "specify sample length (must be greater than n)")
+	csv := flag.String("csv", "", "specify an input file of frequency amplitudes (dB transformed)")
+	out := flag.String("o", "", "specify output file for plot")
+	width := flag.Float64("width", 16, "specify plot width (cm)")
```

Comment:
perhaps use [font.ParseLength](https://pkg.go.dev/gonum.org/v1/plot@v0.9.0/font#ParseLength) to allow for nicer units handling? (_e.g.:_ `42cm`).

Reply:
The documentation becomes unwieldy if I do that (I need to explain what units are possible and what absent units means).

Reply:
ok.